service-base/src/test/java/org/eclipse/hono/service/metric/MicrometerBasedMetricsTest.java

```diff
@@ -218,4 +219,99 @@ public void testPayloadSizeForCommandMessages(final MeterRegistry registry) {
+     *
+     * @param registry The registry that the tests should be run against.
+     */
+    @ParameterizedTest
+    @MethodSource("registries")
+    public void testAuthenticatedConnections(final MeterRegistry registry) {
+
+        final MicrometerBasedMetrics metrics = new MicrometerBasedMetrics(registry);
+
+        final String tenant1 = "tenant1";
+        final String tenant2 = "tenant2";
+
+        // GIVEN a metrics instance that has recorded one telemetry message and one event per tenant
+        metrics.incrementConnections(tenant1);
+        metrics.incrementConnections(tenant1);
+        metrics.incrementConnections(tenant2);
+        metrics.incrementConnections(tenant2);
+
+        final Search search = registry.find(MicrometerBasedMetrics.METER_CONNECTIONS_AUTHENTICATED);
+        assertEquals(2, search.meters().size());
+        assertEquals(4, metrics.getNumberOfConnections());
+
+        // WHEN closing all connections for one tenant
+        metrics.decrementConnections(tenant1);
+        assertEquals(2, search.meters().size());
+        metrics.decrementConnections(tenant1);
+
+        // THEN the metric for this tenant is removed
+        assertEquals(1, search.meters().size());
+        assertEquals(2, metrics.getNumberOfConnections());
+
+    }
+
+    /**
+     * Verifies the behavior of removeTelemetryMetricsForTenant(), i.e. checks that metrics are removed for the given
+     * event type and tenant.
+     *
+     * @param registry The registry that the tests should be run against.
+     */
+    @ParameterizedTest
+    @MethodSource("registries")
+    public void testRemoveTelemetryMetricsForTenant(final MeterRegistry registry) {
+
+        final MicrometerBasedMetrics metrics = new MicrometerBasedMetrics(registry);
+        final String tenant1 = "tenant1";
+        final Tags tenantTag1 = Tags.of(MetricsTags.getTenantTag(tenant1));
+        final String tenant2 = "tenant2";
+        final Tags tenantTag2 = Tags.of(MetricsTags.getTenantTag(tenant2));
+
+        // GIVEN a metrics instance that has recorded one telemetry message and one event per tenant
+        reportTelemetry(metrics, EndpointType.TELEMETRY, tenant1);
+        reportTelemetry(metrics, EndpointType.EVENT, tenant1);
+        reportTelemetry(metrics, EndpointType.TELEMETRY, tenant2);
+        reportTelemetry(metrics, EndpointType.EVENT, tenant2);
+
+        assertEquals(4,
+                registry.find(MicrometerBasedMetrics.METER_MESSAGES_RECEIVED).meters().size());
+        assertEquals(4,
+                registry.find(MicrometerBasedMetrics.METER_MESSAGES_PAYLOAD).meters().size());
+
+        // WHEN removing the telemetry metrics for tenant1
+        metrics.removeTelemetryMetricsForTenant(EndpointType.TELEMETRY, tenant1);
+
+        // THEN only the metric for telemetry and this tenant is removed
+        assertEquals(1,
+                registry.find(MicrometerBasedMetrics.METER_MESSAGES_RECEIVED).tags(tenantTag1).meters().size());
```

Comment:
IMHO the search needs to also include the message type tag to make sure that the right metric has been removed ...